test/1_stdlib/Float.swift

```diff
@@ -222,13 +222,24 @@ func checkQNaN(_ qnan: TestFloat) {
   _precondition(qnan.floatingPointClass == .quietNaN)
 }
 
+func checkSNaN(_ snan: TestFloat) {
+  checkNaN(snan)
+// FIXME: i386 doesn't fully support signaling NaN.
```

Comment:
> On May 13, 2016, at 12:10 PM, Joe Groff <notifications@github.com> wrote:
> 
> In test/1_stdlib/Float.swift <https://github.com/apple/swift/pull/2494#discussion_r63209520>:
> 
> > @@ -222,13 +222,24 @@ func checkQNaN(_ qnan: TestFloat) {
> >    _precondition(qnan.floatingPointClass == .quietNaN)
> >  }
> >  
> > +func checkSNaN(_ snan: TestFloat) {
> > +  checkNaN(snan)
> > +// FIXME: i386 doesn't fully support signaling NaN.
> We do own Swift's calling convention. If it were worth it, we could consider changing the convention, for Float80 on x86_64 at least, to pass x87 floats by memory.
> 
> 

It doesn’t effect Float80 load/store, only Float and Double[*], so there’s no problem on x86_64.

The problem is only on i386, and only for Float and Double.  If we wanted to fix it, we would want to return them in xmm0 instead of fp(0).

[*] The history here is interesting.  x87 originally did not have load/store of 80-bit values, only 32b and 64b, which were extended to 80b on load and rounded on store.  When 80b load/store were added, they were the widest path to memory, so folks wanted to use it for memcpy, which meant that it could not change the values (and hence could not quiet signaling NaNs).